examples/gno.land/p/demo/zteams/teams.gno

```diff
@@ -0,0 +1,262 @@
+    }
+
+    delete(team.Admins, userAddress.String())
+    tm.Teams.Set(teamAddress.String(), team)
+    return nil
+}
+
+// Assigns disburse permission to a user; only the Owner or Admin can do this.
+func (tm *TeamManager) AssignDisbursePermission(teamAddress std.Address, userAddress std.Address, requester std.Address) error {
+    team, err := tm.getTeam(teamAddress)
+    if err != nil {
+        return err
+    }
+
+    // Only Owner or Admin can assign disburse permissions.
+    if !tm.isOwner(team, requester) && !tm.isAdmin(team, requester) {
+        return ErrNotAuthorized
+    }
+
+    // Assign disburse permission.
+    team.Disbursers[userAddress.String()] = true
+    tm.Teams.Set(teamAddress.String(), team)
+    return nil
+}
+
+// Removes disburse permission from a user; only the Owner or Admin can do this.
+func (tm *TeamManager) RemoveDisbursePermission(teamAddress std.Address, userAddress std.Address, requester std.Address) error {
+    team, err := tm.getTeam(teamAddress)
+    if err != nil {
+        return err
+    }
+
+    // Only Owner or Admin can remove disburse permissions.
+    if !tm.isOwner(team, requester) && !tm.isAdmin(team, requester) {
+        return ErrNotAuthorized
+    }
+
+    // Check if the user has disburse permission.
+    if !team.Disbursers[userAddress.String()] {
+        return ErrUserNotDisburser // Use a specific error message.
+    }
+
+    // Remove disburse permission.
+    delete(team.Disbursers, userAddress.String())
+    tm.Teams.Set(teamAddress.String(), team)
+    return nil
+}
+
+// Checks if a user can disburse (either Owner, Admin, or a designated Disburser).
+func (tm *TeamManager) CanDisburse(teamAddress std.Address, userAddress std.Address) bool {
+    team, err := tm.getTeam(teamAddress)
+    if err != nil {
+        return false
+    }
+    return tm.isOwner(team, userAddress) || tm.isAdmin(team, userAddress) || team.Disbursers[userAddress.String()]
+}
+
+// Adds a team to the TeamManager.
+func (tm *TeamManager) AddTeam(t *Team) error {
+    _, found := tm.Teams.Get(t.Address.String())
+    if found {
+        return ErrTeamAddressAlreadyExists
+    }
+
+    // Initialize Admins and Disbursers maps.
+    t.Admins = make(map[string]bool)
+    t.Disbursers = make(map[string]bool)
+
+    tm.Teams.Set(t.Address.String(), t)
+    return nil
+}
+
+// Adds a user to a team; only the Owner or Admin can do this.
+func (tm *TeamManager) AddUserToTeam(teamAddress std.Address, user users.User, requester std.Address) error {
+    team, err := tm.getTeam(teamAddress)
+    if err != nil {
+        return err
+    }
+
+    if !tm.isOwner(team, requester) && !tm.isAdmin(team, requester) {
+        return ErrNotAuthorized
+    }
+
+    // Retrieve current members or initialize an empty slice if nil.
+    members, _ := tm.Members.Get(teamAddress.String())
+    memberList, ok := members.([]*users.User)
+    if !ok || memberList == nil {
+        memberList = []*users.User{}
+    }
```

Comment:
How about extracting this common logic in `AddUserToTeam` and `RemoveUserFromTeam` functions into a separate function?

Reply:
The logic is not quite common, it is slightly different: in one case I'm returning an empty slice (for adding an user), in the other one I'm returning an error for user not being found. Abstracting this into a separate function, with 2 return types, would probably be more confusing (at least for me).

Let me know your thoughts on this one.